modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -65,22 +65,39 @@ def _unstrip_protocol(protocol, path):
 
         if isinstance(path, AbstractBufferedFile):
             return path.fs, [path]
-        protocol, path = split_protocol(path)
-        filesystem = get_filesystem_class(protocol)(**storage_options)
-        if filesystem.stat(path)["type"] == "directory":
-            files = []
-            for path in sorted(filesystem.find(path)):
-                if version.parse(fsspec.__version__) < version.parse("2022.5.0"):
-                    files.append(_unstrip_protocol(filesystem.protocol, path))
-                else:
-                    files.append(filesystem.unstrip_protocol(path))
-            return filesystem, files
-
-        return filesystem, [
-            _unstrip_protocol(filesystem.protocol, path)
-            if version.parse(fsspec.__version__) < version.parse("2022.5.0")
-            else filesystem.unstrip_protocol(path)
-        ]
+        filesystem = get_filesystem_class(split_protocol(path)[0])(**storage_options)
+        dataset = cls._get_dataset(path, storage_options)
+        files = []
+        for fpath in dataset.files:
+            if version.parse(fsspec.__version__) < version.parse("2022.5.0"):
```

Comment:
please swap `fsspec` version check and `for` loop - it's better to do two loops, one in `if` and other in `else`, this would work faster

Reply:
Oh right this is a good point. I looked into the implementation of parse and it's a ton of regex matching which is usually pretty expensive.  Thanks for the suggestion!